src/pri-dual-vec.md

```diff
@@ -0,0 +1,193 @@
+# Primal and Dual Vectors
+
+The finite element method uses vectors of data in a variety of ways and the
+differences can be subtle.  MFEM defines `GridFunction`, `LinearForm`, and
+`Vector` classes which help to distinguish the different roles that vectors of
+data can play.
+
+<center>**Graphical summary of Primal, Dual, DoF, True DoF vectors**</center>
+![](img/FEM-LA-Diagrams.png)
+
+## Primal Vectors
+
+The finite element method is based on the notion that a smooth function can be
+approximated by a sum of piece-wise functions called *basis functions*:
+$$f(\vec{x})\approx\sum_i f_i \phi_i(\vec{x}) \label{expan}$$
+The support of an individual basis function, $\;\phi_i(\vec{x})$, will either be
+a single zone or a collection of zones that share a common vertex, edge, or
+face.  The expansion coefficients, $\;f_i$, are linear functionals of the field
+being approximated, $\;f(\vec{x})$ in this case.  The $\;f_i$ could be as simple
+as values of the function at particular points, called interpolation points,
+e.g. $\;f_i=f(\vec{x}\_i)$ or they could be integrals of the field over
+submanifolds of the domain, e.g. $\;f_i = \int_{\Omega_i}f(\vec{x})d\vec{x}$.
+There are many possibilities but the expansion coefficients must be linear
+functionals of $\;f(\vec{x})$.  The expansion coefficients are often called
+*degrees of freedom*, or *DoFs* for short, though in certain cases they may
+not be actually independent.  We'll discuss this more in a later
+section on [True DoFs](pri-dual-vec.md#true-degree-of-freedom-vectors).
+
+Once the basis functions are defined, with some unique ordering, the expansion
+coefficients can be stored in a vector using the same order.  Such a vector of
+coefficients is called a *primal vector*.  The original function,
+$\;f(\vec{x})$, can then be approximated using \eqref{expan}.  In practice this
+requires not only the primal vector of coefficients but also knowledge of the
+mesh and the basis functions for each element of the mesh.  In MFEM these
+collections of information are combined into `GridFunction` objects (or
+`ParGridFunction` objects when used in parallel).
+
+The `GridFunction` object can be used to compute a primal vector using any of
```

Comment:
I think it would be more clear to first talk about the `Get` methods since they are used to evaluate the `GridFunction` at certain points, which is a core feature of a function I think. Whereas `Project` is used to project a GridFunction on a different functional space, resulting in a different primal vector (associated with the new basis of the potential new functional space, *e.g* discontinuous to continuous functional space projection).

Reply:
You are correct, of course.  I always think about these in the reverse order because you cannot call the `Get` methods until you've projected something onto the `GridFunction`.  I'll try to reorder this.